guarddog/scanners/project_scanner.py

```diff
@@ -97,8 +98,22 @@ def versions(package_name):
 
         dependencies = {}
 
+        def safe_parse_requirements(req):
+            parsed = pkg_resources.parse_requirements(req)
+            while True:
```

Comment:
any specific reason we'd not just use a for loop on the generator instead of a `while true + except on StopIteration`?

Reply:
if we do `for x in pkg_resources.parse_requirements(req)` we won't be able to catch the exception thrown in the implicit `next`

Reply:
Just tested with 
```
        def safe_parse_requirements(req):
            try:
                for parsed in pkg_resources.parse_requirements(req):
                    yield parsed
            except Exception as e:
                sys.stderr.write(f"Error when parsing requirements, received error {str(e)}. This entry will be "
                                 "ignored.\n")
                yield None
```
it seems it stops on te first error (i.e. the result in the test does not have the `garddog` entry)